document-store/src/main/java/org/hypertrace/core/documentstore/postgres/PostgresCollection.java

```diff
@@ -263,8 +269,166 @@ public BulkUpdateResult bulkUpdateSubDocs(Map<Key, Map<String, Document>> docume
+        for (String pathToken : pathTokens) {
+          currNode = currNode.path(pathToken);
+        }
+        // if we do not end up in an array node, then the path doesn't exist. So do not do anything
+        if (!currNode.isArray()) {
+          continue;
+        }
+        ArrayNode candidateArray = (ArrayNode) currNode;
+        for (Document subDoc : request.getSubDocuments()) {
+          Iterator<JsonNode> iterator = candidateArray.iterator();
+          List<Integer> indicesToRemove = new ArrayList<>();
+          int idx = 0;
+          while (iterator.hasNext()) {
+            if (iterator.next().equals(MAPPER.readTree(subDoc.toJson()))) {
+              indicesToRemove.add(idx);
+            }
+            idx++;
+          }
+          indicesToRemove.forEach(candidateArray::remove);
+        }
+        upsertMap.put(key, new JSONDocument(rootNode));
+      } else {
+        LOGGER.warn("Could not get row with key: {}", key);
+      }
+    }
+    return upsertDocs(upsertMap);
+  }
+
+  private Query getSelectDocumentQueryForKey(Key key) {
+    return new Query().withSelection(DOCUMENT).withFilter(Filter.eq(ID, key.toString()));
+  }
+
+  private BulkUpdateResult addImpl(BulkArrayValueUpdateRequest request) throws IOException {
+    Set<Key> keys = request.getKeys();
+    Map<Key, Document> upsertMap = new HashMap<>();
+    for (Key key : keys) {
+      Query getQuery = getSelectDocumentQueryForKey(key);
+      CloseableIterator<Document> searchRes = search(getQuery);
+      if (searchRes.hasNext()) {
+        JsonNode rootNode;
+        try {
+          Document doc = searchRes.next();
+          rootNode = MAPPER.readTree(doc.toJson());
+        } catch (JsonProcessingException e) {
+          LOGGER.warn("Malformed doc for id: {}, continuing for remaining keys", key, e);
+          continue;
+        }
+        // create path if missing
+        JsonNode arrayNode = createPathInDoc(request.getSubDocPath(), rootNode);
+        ArrayNode candidateArray = (ArrayNode) arrayNode;
+        for (Document subDoc : request.getSubDocuments()) {
+          // we don't insert docs if they exist already (so the array behaves like a set here)
+          boolean alreadyContainsDoc = false;
+          Iterator<JsonNode> arrayElems = candidateArray.elements();
+          JsonNode subDocAsJSON = getSubDocAsJSON(key, subDoc);
+          while (arrayElems.hasNext()) {
+            JsonNode next = arrayElems.next();
+            if (next.equals(subDocAsJSON)) {
+              alreadyContainsDoc = true;
+              break;
+            }
+          }
+          if (!alreadyContainsDoc) {
+            candidateArray.add(subDocAsJSON);
+          }
+        }
+        upsertMap.put(key, new JSONDocument(rootNode));
+      } else {
+        LOGGER.warn("Could not get row with key: {}", key);
+      }
+    }
+    return upsertDocs(upsertMap);
+  }
+
+  private JsonNode getSubDocAsJSON(Key key, Document subDoc) throws IOException {
```

Comment:
Instead of throwing, would returning null will suffice? So, that way places where you want to continue, will be avoided.

```
private JsonNode convertDocToJson(Document subDoc) {
    JsonNode subDocAsJSON;
    try {
      subDocAsJSON = MAPPER.readTree(subDoc.toJson());
    } catch (Exception e) {
      LOGGER.error("Malformed JSON", e);
      return null;
    }
    return subDocAsJSON;
  }
```

Reply:
@kotharironak Actually did this to keep the behaviour similar to Mongo. Mongo currently throws a `JSONParseException` if the docs are malformed.